syntax = "proto3";

message Compiler{
  int64 id = 1;
  string name = 2;
  string extend = 3;
}

message GetListCompilerRequest{
  int64 count = 1;
}

message GetListCompilerResponse{
  repeated Compiler compilers = 1;
}

message GetCompilerRequest{
  int64 id = 1;
}

service CompilerApi{
  rpc GetListCompiler(GetListCompilerRequest) returns (GetListCompilerResponse){}
  rpc GetCompiler(GetCompilerRequest) returns (Compiler){}
}